src/main/scala/scalaj/http/Http.scala

```diff
@@ -541,6 +549,30 @@ case class StringBodyConnectFunc(data: String) extends Function2[HttpRequest, Ht
   override def toString = "StringBodyConnectFunc(" + data + ")"
 }
 
+case class InputStreamBodyConnectFunc(inputStream: InputStream) extends Function2[HttpRequest, HttpURLConnection, Unit] {
+  def apply(req: HttpRequest, conn: HttpURLConnection): Unit = {
+    conn.setDoOutput(true)
+    conn.connect()
+
+    recursive(new Array[Byte](req.sendBufferSize), conn.getOutputStream, 0L)
+  }
+
+  @tailrec
+  private def recursive(buffer: Array[Byte], out: OutputStream, writtenBytes: Long): Unit = {
```

Comment:
Some another block of code below in `MultiPartConnectFunc` where we're copying all the bytes from one stream to another. Can you break this out int a function that can be used by both?

Reply:
Thanks for the suggestion @hoffrocket. I put the common stuff in a trait (`StreamUtils`) for now and let both `MultiPartConnectFunc` and (the new) `InputStreamBodyConnectFunc` extend from this trait. If you got a better name for this trait, please let me know!

Given the fact that `MultiPartConnectFunc` allows for a callback function (`Long => Unit`), I had to also build that into `StreamUtils.copyStream`. Hence I thought it might be useful to have a similar callback come from the new overrides of `postData` and `put` that were introduced earlier in this pull request. All of them have default arguments `_ => ()`.